### ISSUE TITLE
ldc 1.24.0 has stopped building on nixpkgs unstable

### ISSUE THREAD

**Issue by ralph-amissah** [NixOS/nixpkgs#114875]
**ldc 1.24.0 has stopped building on nixpkgs unstable**

the most recent version of ldc made available in nixpkgs fails to build
This version used to build, but has stopped doing so

**To Reproduce**
1. nix-channel --list # unstable https://nixos.org/channels/nixpkgs-unstable
2. nix-channel --update
3. nix-shell -p ldc

**Expected behavior**
ldc should build and be available

**Additional context**
I reproduce this behavior on all my nix boxen;
it is probably easiest to extract extensive build error messages on own system (they should be readily available).
If this assumption is incorrect, I will gladly provide further feedback.

**Notify maintainers**
cc @ThomasMader @lionello 

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 5.11.1, NixOS, 21.05pre273435.0aeba64fb26 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20201205_a5d85d0`
 - channels(ralph): `"unstable-21.05pre272788.870dbb751f4"`
 - channels(root): `"nixos-21.05pre273435.0aeba64fb26"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- ldc
# a list of nixos modules affected by the problem
module:
```

**Comment by i077** [comment]
I think this is because ldc 1.24.0 (which is out of date btw) does not support LLVM 11.1. Bisecting nixpkgs shows an update to LLVM 11.1.0 (commit 63037b0348f23551639051282453dc02189c1d68, PR #110631) causes this, which I think confirms it. Note that v1.25.0+ (release notes [here](https://github.com/ldc-developers/ldc/releases)) explicitly adds this support for the newer LLVM.

I did try updating ldc to 1.25.1, but using `nixpkgs-review wip` I found that this causes dmd (and a handful of other packages) to fail. dmd also happens to be out of date on nixpkgs, but there are several patches applied to it that I'm not sure how to deal with for the latest version.

cc @ThomasMader @lionello here for dmd as well as ldc.

**Comment by ralph-amissah** [comment]
Thank you for your response.

I was waiting for ldc to work again before I sent a bug report for dmd.

(I think you will find that) **dmd has not built on nixpkgs unstable for months**, and ldc is not used in its build so it seems unlikely that you introduce any new problem here. At least this has been the case for me, and is the reason that I have had to modify dub's default.nix locally to build with ldc (ldmd2) if only that is made available to it in shell.nix That is a fairly minor modification hat may be useful more generally and perhaps I should share that too.

I will see what happens when I try to update to v1.25.0.

To sum up, without having looked yet, I hazard the guess that you are not further breaking dmd (which is already broken) by updating ldc to 1.25.1; and if updating ldc to 1.25.1 works, it may indeed be all that is needed to fix this particular problem, i.e. ldc building.

cc @ThomasMader @lionello here for dmd as well as ldc.

**Comment by FliegendeWurst** [comment]
This was fixed by #115265
